src/rng/backends/rocrand/rocrand_task.hpp

```diff
@@ -36,7 +36,11 @@ static inline void host_task_internal(H &cgh, E e, F f) {
 #else
 template <typename H, typename A, typename E, typename F>
 static inline void host_task_internal(H &cgh, A acc, E e, F f) {
+#ifdef SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND
+    cgh.ext_codeplay_enqueue_native_command([=](sycl::interop_handle ih){
```

Comment:
We should also make sure to avoid calling `hipStreamSynchronize` a few lines below when `SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND` is defined.
Same pattern in the other overload of `host_task_internal` below.

Reply:
I'm concerned because I don't see `cuStreamSynchronize` being called for curand.

Reply:
> We should also make sure to avoid calling `hipStreamSynchronize` a few lines below when `SYCL_EXT_ONEAPI_ENQUEUE_NATIVE_COMMAND` is defined. Same pattern in the other overload of `host_task_internal` below.

My mistake, I missed that. Thanks

Reply:
> I'm concerned because I don't see `cuStreamSynchronize` being called for curand.

OK, I'll fix the existing impl as well.

Reply:
I've updated this now. I've updated tests with the host_task path also since the PR now affects that path too.